feat: add `trip_id` to `vehicle_trips` tables

rymarczy opened the pull request:
This is a first attempt at adding `trip_id` to the `vehicle_trips` table.

This methodology can be refined by a future PR, but `trip_id` is required to move forward with updated `static_trip_id_guess` matching need for OPMI QA improvements.

Asana Task: https://app.asana.com/0/1204389777951216/1204505200416990
Asana Task: https://app.asana.com/0/1204517240145658/1204119636952116

rymarczy commented on python_src/tests/test_files/april_2023_filepaths.json:
new seed files needed to test RT vehicle events with valid `trip_id` data

rymarczy commented on python_src/src/lamp_py/postgres/postgres_utils.py:
```diff
@@ -266,16 +266,29 @@ def select_as_list(
+    def truncate_table(
+        self,
+        table_to_truncate: Any,
+        restart_identity: bool = False,
+        cascade: bool = False,
+    ) -> None:
         """
         truncate db table
 
-        sqlalchemy has no truncate operation so this is the closest equivalent
+        restart_identity: Automatically restart sequences owned by columns of the truncated table(s).
+        cascade: Automatically truncate all tables that have foreign-key references to any of the named tables, or to any tables added to the group due to CASCADE.
         """
         truncat_as = self._get_schema_table(table_to_truncate)
 
-        truncat_as.drop(self.engine)
-        truncat_as.create(self.engine)
+        truncate_query = f"TRUNCATE {truncat_as}"
+
+        if restart_identity:
+            truncate_query = f"{truncate_query} RESTART IDENTITY"
+
+        if cascade:
+            truncate_query = f"{truncate_query} CASCADE"
+
+        self.execute(sa.text(f"{truncate_query};"))
```
Updated truncate function that doesn't drop and re-create whole table. Runs standard postgress truncate command.

rymarczy commented on python_src/src/lamp_py/performance_manager/l0_gtfs_rt_events.py:
```diff
@@ -108,6 +110,7 @@ def combine_events(
         "stop_sequence",
         "trip_stop_hash",
         "vehicle_id",
+        "trip_id",
```
Pull `trip_id` from parquet files

rymarczy commented on python_src/src/lamp_py/performance_manager/l0_rt_trip_updates.py:
```diff
@@ -30,6 +30,7 @@ def get_tu_dataframe_chunks(
         "start_date",
         "start_time",
         "vehicle_id",
+        "trip_id",
```
Pull `trip_id` from parquet files

rymarczy commented on python_src/src/lamp_py/performance_manager/l1_rt_trips.py:
```diff
@@ -17,14 +18,31 @@
 
-def insert_trips_hash_columns(
+def load_temp_for_hash_compare(
     db_manager: DatabaseManager, events: pandas.DataFrame
 ) -> None:
     """
-    Load new 'trip_hash' records and columns into RDS
+    Load "temp_hash_compare" table with "trip_hash" from newly inserted RT events
+    """
+    new_trip_hash_bytes = events.loc[
+        events["do_insert"], "trip_hash"
+    ].drop_duplicates()
+
+    db_manager.truncate_table(TempHashCompare)
+    db_manager.execute_with_data(
+        sa.insert(TempHashCompare.__table__),
+        new_trip_hash_bytes.to_frame("trip_stop_hash"),
+    )
```
Broke up this process into a few smaller functions

rymarczy commented on python_src/src/lamp_py/performance_manager/l1_rt_trips.py:
```diff
@@ -17,14 +18,31 @@
+    )
+
+
+def load_new_trip_data(
+    db_manager: DatabaseManager, events: pandas.DataFrame
+) -> None:
+    """
```
This works the same as before, just with slightly less scope in this function and the addition of `trip_id`

rymarczy commented on python_src/src/lamp_py/performance_manager/l1_rt_trips.py:
```diff
@@ -85,6 +96,54 @@ def insert_trips_hash_columns(
         )
 
 
+def update_trip_stop_counts(db_manager: DatabaseManager) -> None:
+    """
+    Update "stop_count" field based on "trip_hash"s with new events
+    """
+    new_stop_counts_cte = (
+        sa.select(
+            VehicleEvents.trip_hash,
+            sa.func.count(VehicleEvents.trip_stop_hash).label("stop_count"),
+        )
+        .select_from(VehicleEvents)
+        .join(
+            TempHashCompare,
+            TempHashCompare.trip_stop_hash == VehicleEvents.trip_hash,
+        )
+        .where(
+            sa.or_(
+                VehicleEvents.vp_move_timestamp.is_not(None),
+                VehicleEvents.vp_stop_timestamp.is_not(None),
+            ),
+        )
+        .group_by(
+            VehicleEvents.trip_hash,
+        )
+        .cte("new_stop_counts")
+    )
+
+    update_query = (
+        sa.update(VehicleTrips.__table__)
+        .where(
+            VehicleTrips.trip_hash == new_stop_counts_cte.c.trip_hash,
+        )
+        .values(stop_count=new_stop_counts_cte.c.stop_count)
+    )
+
+    db_manager.execute(update_query)
```
Process updates to `stop_count` in the `vehicle_trips` table separately.

rymarczy commented on python_src/src/lamp_py/performance_manager/l0_rt_vehicle_positions.py:
```diff
@@ -136,6 +137,8 @@ def transform_vp_timestamps(
             "vehicle_id",
         ],
     )
+    # TODO: review trip_id processing # pylint: disable=fixme
+    #  add more intelligent trip_id processing, this approach will randomly select trip_id record to keep
```
The logic we are using to add `trip_id` is slightly random, it is functional but should be reviewed in the future to see exactly what `trip_id` we are pulling compared to the `trip_hash` grouping we are doing separately.

mzappitello reviewed (commented):
a few questions before merging.

rymarczy commented:
Incorporated the following comments: 
- added new seed metadata files to docker-compose.yml
- renamed `trip_stop_hash` in the `temp_hash_compare` table to `hash`

mzappitello commented:
LGTM
